packages/web3-errors/src/errors/rpc_errors.ts

```diff
@@ -0,0 +1,176 @@
+		super(rpcError, 'Parse error');
+	}
+}
+
+export class InvalidRequestError extends RpcError {
+	public code = ERR_RPC_INVALID_REQUEST;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Invalid request');
+	}
+}
+
+export class MethodNotFoundError extends RpcError {
+	public code = ERR_RPC_INVALID_METHOD;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Method not found');
+	}
+}
+
+export class InvalidParamsError extends RpcError {
+	public code = ERR_RPC_INVALID_PARAMS;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Invalid request');
+	}
+}
+
+export class InternalError extends RpcError {
+	public code = ERR_RPC_INTERNAL_ERROR;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Internal error');
+	}
+}
+
+export class InvalidInputError extends RpcError {
+	public code = ERR_RPC_INVALID_INPUT;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Invalid input');
+	}
+}
+
+export class MethodNotSupported extends RpcError {
+	public code = ERR_RPC_UNSUPPORTED_METHOD;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Method not supported');
+	}
+}
+
+export class ResourceUnavailableError extends RpcError {
+	public code = ERR_RPC_UNAVAILABLE_RESOURCE;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Resource unavailable');
+	}
+}
+
+export class ResourcesNotFoundError extends RpcError {
+	public code = ERR_RPC_MISSING_RESOURCE;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Resource not found');
+	}
+}
+
+export class VersionNotSupportedError extends RpcError {
+	public code = ERR_RPC_NOT_SUPPORTED;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'JSON-RPC version not supported');
+	}
+}
+
+export class TransactionRejectedError extends RpcError {
+	public code = ERR_RPC_TRANSACTION_REJECTED;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Transaction rejected');
+	}
+}
+
+export class LimitExceededError extends RpcError {
+	public code = ERR_RPC_LIMIT_EXCEEDED;
+	public constructor(rpcError: JsonRpcResponseWithError) {
+		super(rpcError, 'Limit exceeded');
+	}
+}
+
+export const rpcErrorValues: { [k: string]: { message: string } } = {
+	'-32700': {
+		message: 'Parse error',
+	},
+	'-32600': {
+		message: 'Invalid request',
+	},
+	'-32601': {
+		message: 'Method not found',
+	},
+	'-32602': {
+		message: 'Invalid params',
+	},
+	'-32603': {
+		message: 'Internal Error',
+	},
+	'-32000': {
+		message: 'Invalid input',
+	},
+	'-32001': {
+		message: 'Resource not found',
+	},
+	'-32002': {
+		message: 'Resource unavailable',
+	},
+	'-32003': {
+		message: 'Transaction rejected',
+	},
+	'-32004': {
+		message: 'Method not supported',
+	},
+	'-32005': {
+		message: 'Limit exceeded',
+	},
+	'-32006': {
```

Comment:
I think create dictionary/map with directly using existing error codes ( for avoiding any error code duplication and future maintenance) . 

```
    export const rpcErrorsMap = new Map<number, object>();
    rpcErrorsMap.set(ERR_RPC_INVALID_JSON, ParseError);
    rpcErrorsMap.set(ERR_RPC_INVALID_REQUEST, InvalidRequestError);
    //.....
```